src/pfsense_api_enhanced.py

```
@@ -123,62 +128,80 @@ def __init__(
+        """Ensure HTTP client is created for current event loop"""
+        try:
+            current_loop = asyncio.get_running_loop()
+        except RuntimeError:
+            current_loop = None
+
+        # Recreate client if loop changed or client doesn't exist
+        if self.client is None or self._client_loop != current_loop:
+            if self.client is not None:
+                # Close old client if it exists
+                try:
+                    asyncio.create_task(self.client.aclose())
```

Ruff (0.15.0)

[warning] Store a reference to the return value of `asyncio.create_task`

(RUF006)